### PR TITLE
Consolidate dirs which doesn't ve CONSOLIDATED flag

### PR DESCRIPTION
**Pull request by gsriram7**
@sriram-r @ashwanthkumar

### CODE REVIEW

File: dfs-datastores/src/main/java/com/backtype/hadoop/pail/Pail.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import java.util.*;
 
 public class Pail<T> extends AbstractPail implements Iterable<T>{
+    private static final String CONSOLIDATED = "CONSOLIDATED";
```
Comment (sriram-r) [inline dfs-datastores/src/main/java/com/backtype/hadoop/pail/Pail.java]: Please reuse these constants.

File: dfs-datastores/src/main/java/com/backtype/hadoop/pail/Pail.java [hunk 2]
```diff
@@ -527,6 +528,14 @@ public void consolidate(long maxSize) throws IOException {
         Consolidator.consolidate(_fs, _format, new PailPathLister(false), consolidatedirs, maxSize, EXTENSION, structure, getRoot());
     }
 
+    private boolean isConsolidated(String dir) throws IOException {
+        try {
+            return _fs.exists(new Path(dir + "/" + CONSOLIDATED));
```
Comment (sriram-r) [inline dfs-datastores/src/main/java/com/backtype/hadoop/pail/Pail.java]: Does `fs.exists` throw an `FileNotFoundException`? It returns only a boolean. It throws an IOException only when there's stream related errors.
```
    public boolean exists(Path f) throws IOException {
        try {
            return this.getFileStatus(f) != null;
        } catch (FileNotFoundException var3) {
            return false;
        }
    }
```

Comment (ashwanthkumar) [inline dfs-datastores/src/main/java/com/backtype/hadoop/pail/Pail.java]: Can we re-use the existing `exists()` method instead of `_fs.exists`?

**Comment by sriram-r** [comment]
@gsriram7 Can you add a test to assert the presence of CONSOLIDATED file in the target dir?

**Comment by ashwanthkumar** [comment]
@gsriram7 Also can  you please explain the context on why we're doing this as part of the commit? Since this is an Open source project, it would help outside folks (read it as forks) get more context.

**Comment by gsriram7** [comment]
When files are consolidated, they are deleted from s3. But the next runs of consolidator picks up the deleted files while enlisting all the files in the s3 location due to s3's eventual consistency for deletes. So the idea is to maintain a flag "CONSOLIDATED" inside the s3 partition so that only the partitions without the flag will be listed. The solution wont work because when the same directory has to be consolidated more than once eg: products/master, the consolidation will be skipped as the first run will set the flag and subsequent runs will ignore the directories.

**Pull request closed without merging**